src/dmd/mars.d

```diff
@@ -1345,6 +1362,70 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
         return true;
     }
 
+    /**
+     * Trigger an error messsage about an invalid switch.
+     * If an optional supplemental message has been provided,
+     * it will be printed too
+     *
+     * Params:
+     *  p = 0 terminated string
+     *  s = supplemental help message listing the available options
+     */
+    void errorInvalidSwitch(const(char)* p, string availableOptions = null)
+    {
+        error("Switch `%s` is invalid", p);
+        if (availableOptions !is null)
+            errorSupplemental(Loc.initial, "%.*s", availableOptions.length, availableOptions.ptr);
```

Comment:
All `.length' arguments to `%.*s` need to be cast to `int`, because `.length` is a `size_t`.
This is a global comment. Please check all `%.*s` arguments.

Reply:
https://github.com/dlang/dmd/pull/9258